src/server/pps/server/server.go

```diff
@@ -45,12 +48,15 @@ type Env struct {
 }
 
 func EnvFromServiceEnv(senv serviceenv.ServiceEnv, txnEnv *txnenv.TransactionEnv, reporter *metrics.Reporter) Env {
+	etcdPrefix := path.Join(senv.Config().EtcdPrefix, senv.Config().PPSEtcdPrefix)
 	return Env{
 		DB:            senv.GetDBClient(),
 		TxnEnv:        txnEnv,
 		Listener:      senv.GetPostgresListener(),
 		KubeClient:    senv.GetKubeClient(),
 		EtcdClient:    senv.GetEtcdClient(),
+		EtcdPrefix:    etcdPrefix,
+		TaskService:   senv.GetTaskService(etcdPrefix),
```

Comment:
Do we actually need the etcdClient for anything other than the task service?

Reply:
Yeah, for the code that connects to the worker API servers. The connection info is recorded in etcd by the workers.